src/utils/parsers.rs

```diff
@@ -20,9 +22,43 @@ pub fn parse_schema_source(loc: &str) -> Result<SchemaSource> {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct GraphRef {
+    pub name: String,
+    pub variant: String,
+}
+
+/// NOTE: THIS IS A TEMPORARY SOLUTION. IN THE FUTURE, ALL GRAPH ID PARSING
+/// WILL HAPPEN IN THE BACKEND TO KEEP EVERYTHING CONSISTENT. THIS IS AN
+/// INCOMPLETE PLACEHOLDER, AND MAY NOT COVER EVERY SINGLE VALID USE CASE
+pub fn parse_graph_ref(graph_id: &str) -> Result<GraphRef> {
+    let pattern = Regex::new(r"^[a-zA-Z][a-zA-Z0-9_-]{0,63}$").unwrap();
```

Comment:
This variable here is for the `<NAME>` portion of the graph id, yea? Would it be more useful to name this `name_pattern`?